test/metric_value_benchmark/swap_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"log"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type SwapTestRunner struct {
+	BaseTestRunner
+}
+
+var _ ITestRunner = (*SwapTestRunner)(nil)
+
+func (t *SwapTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := t.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = t.validateSwapMetric(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *SwapTestRunner) getTestName() string {
+	return "Swap"
+}
+
+func (t *SwapTestRunner) getAgentConfigFileName() string {
+	return "swap_config.json"
+}
+func (t *SwapTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
```

Comment:
This can be improved to though, instead of using `sleep` which make the system wasteful in that amount of time, we can do polling interval to validate aggressively.

Reply:
That's not specific to this test. Are you suggesting that we change it in the entire package?

Reply:
Its only improvement and can be done in future version (so not blocking on it). But in a simpler way, changing how we test more aggressively.